### PR TITLE
Plugin YAML Update

### PR DESCRIPTION
**Pull request by pablo67340**
Update plugin YAML containing all Essentials Permission nodes,
descriptions and defaults. This is required by other programs/plugins.

### CODE REVIEW

**Comment by pablo67340** [comment]
Ignore the first commit. I had accidentally pasted all the perms without indentation, thus resulting in a syntax error. I've gone and re-pasted them with the correct indentation required for YAML.

File: Essentials/src/plugin.yml [hunk 1]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+    description: If enabled in the config, this permission allows unsafe enchantments.
+    default: false
+  essentials.give:
+    description: Allow access to the /give command.
+    default: false
+  essentials.itemspawn.exempt:
+    description: Allows spawning of items on the spawn blacklist.
+    default: false
+  essentials.itemspawn.item <itemname>:
+    description: If permission based item spawn: Spawn <itemname.
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: This should have a closing >.

File: Essentials/src/plugin.yml [hunk 2]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+    default: false
+  essentials.itemspawn.item [itemid]:
+    description: If permission based item spawn: Spawn [itemid.
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: Same as above, but with a ].

File: Essentials/src/plugin.yml [hunk 3]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+    description: Allow access to the /hat command.
+    default: false
+  essentials.heal:
+    description: Allow access to the /heal command.
+    default: false
+  essentials.heal.cooldown.bypass:
+    description: Bypass the heal cooldown
+  essentials.heal.others:
+    description: Allows healing another player
+  essentials.itemspawn.meta chapter [chapter]:
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: As far as I can tell, this should have dashes (unless YAML breaks if they are added).

File: Essentials/src/plugin.yml [hunk 4]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+  essentials.enchantments.allowunsafe:
+    description: If enabled in the config, this permission allows unsafe enchantments.
+    default: false
+  essentials.item:
+    description: Allow access to the /item command.
+    default: false
+  essentials.itemspawn.exempt:
+    description: Allows spawning of items on the spawn blacklist.
+    default: false
+  essentials.itemspawn.item [itemid]:
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: Another potentially missing dash and missing closing brackets here.

File: Essentials/src/plugin.yml [hunk 5]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+    default: false
+  essentials.time.set:
+    description: User is allowed to change the time of the world.
+    default: false
+  essentials.tree:
+    description: Allow access to the /tree command.
+    default: false
+  essentials.unlimited:
+    description: Allow access to the /unlimited command.
+    default: false
```
Comment (mdcfe) [inline Essentials/src/plugin.yml:761]: More potentially-missing dashes.

File: Essentials/src/plugin.yml [hunk 6]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+  essentials.seen.banreason:
+    description: Allows the user to see why a user is banned.
+    default: false
+  essentials.seen.extra:
+    description: Allow the user to see the last IP address a user connected with, and their last logged in location.
+    default: false
+  essentials.seen.ipsearch:
+    description: Allows the user to search IPs for username matches.
+    default: false
```
Comment (mdcfe) [inline Essentials/src/plugin.yml:967]: The `essentials.spawner.[mob]` and `essentials.spawner.*` perms are presumably only for `/spawner` and not `/spawnmob`.

File: Essentials/src/plugin.yml [hunk 7]
```diff
@@ -527,3 +527,1183 @@ permissions:
...
+    description: Allows temp banning of players who are offline. This may allow you to ban exempt players.
+    default: false
+  essentials.tempban.unlimited:
+    description: Allows bypassing config file max ban time limit.
+    default: false
+  essentials.thunder:
+    description: Allow access to the /thunder command.
+    default: false
+  essentials.jail.allow.<command>:
+    description: Allows   essentials.commands to be used while jailed (requires regular perm).
```
Comment (mdcfe) [inline Essentials/src/plugin.yml]: Extra spaces here.

**Review by mdcfe — CHANGES_REQUESTED** [review]
I've looked through and left comments on anything I noticed that doesn't look right. Was there a reason for omitting the dashes? They appear to be present on the Essentials wiki page.

**Comment by pablo67340** [comment]
The reason being was I used a few small tricks to actually copy/paste directly from the Wiki. My auto formatter has accidentally taken things out which I've missed. This took me about 30 min to throw together, saved a ton of time. I will fix right now.

**Comment by mdcfe** [comment]
I guessed that they were taken from the wiki :p

I'll check over the PR again in a bit.

**Review by mdcfe — COMMENTED** [review]
Are the added permissions from the EssentialsX changes wiki included here? If so, this looks good.

**Comment by SupaHam** [comment]
I don't think we should be adding permissions that contain parameters. Bukkit will register them as permissions when in fact they are not the real permissions used.

**Comment by pablo67340** [comment]
So shall I remove them entirely?

**Comment by mdcfe** [comment]
I think it would be best to keep the base permissions of the perms with parameters and remove the ones that actually have the parameters.

**Comment by pablo67340** [comment]
There we go, let me know what you think.

**Review by mdcfe — DISMISSED** [review]

**Comment by drtshock** [comment]
Late to the party but why default to false? That will mean that ops don't have permission to do anything.

**Review by DoNotSpamPls — CHANGES_REQUESTED** [review]
The permissions should be default: op
Also, dashes are still missing and I'm not even sure that YAML allows the -[] syntax you've used

**Comment by Ichbinjoe** [comment]
This is likely out of date and dead. Reopen if not.

**Pull request closed without merging**